runtime/src/impls.rs

```diff
@@ -17,24 +17,66 @@
 //! Some configurable implementations as associated type for the substrate runtime.
 
 use crate::constants::fee::TARGET_BLOCK_FULLNESS;
-use crate::{Call, MaximumBlockWeight, Runtime};
+use crate::{Call, MaximumBlockWeight, NegativeImbalance, PositiveImbalance, Runtime};
 use cennznet_primitives::{
 	traits::{BuyFeeAsset, IsGasMeteredCall},
 	types::{Balance, FeeExchange},
 };
 use crml_transaction_payment::GAS_FEE_EXCHANGE_KEY;
 use frame_support::{
+	additional_traits::{InherentAssetIdProvider, MultiCurrencyAccounting},
 	storage,
-	traits::{Currency, ExistenceRequirement, Get, OnUnbalanced, WithdrawReason},
+	traits::{Currency, ExistenceRequirement, Get, Imbalance, OnUnbalanced, WithdrawReason},
 	weights::Weight,
 };
 use pallet_contracts::{Gas, GasMeter};
 use pallet_generic_asset::StakingAssetCurrency;
+use pallet_staking::RewardDestination;
 use sp_runtime::{
 	traits::{CheckedMul, CheckedSub, Convert, SaturatedConversion, Saturating, UniqueSaturatedFrom, Zero},
 	DispatchError, Fixed64,
 };
 
+type CennzxSpot<T> = crml_cennzx_spot::Module<T>;
+type Contracts<T> = pallet_contracts::Module<T>;
+type GenericAsset<T> = pallet_generic_asset::Module<T>;
+type Staking<T> = pallet_staking::Module<T>;
+
+pub struct SplitToAllValidators;
+
+/// This handles the ```NegativeImbalance``` created for transaction fee.
+/// The reward is split evenly and distributed to all of the current elected validators.
+/// The remainder from the division are burned.
+impl OnUnbalanced<NegativeImbalance> for SplitToAllValidators {
+	fn on_nonzero_unbalanced(imbalance: NegativeImbalance) {
+		let validators = Staking::<Runtime>::current_elected();
+		if validators.len().is_zero() || imbalance.peek().is_zero() {
+			return;
+		}
+		// Get a list of elected validators
+		let per_validator_reward: Balance = imbalance.peek() / (validators.len() as Balance);
```

Comment:
```len() as Balance``` is a bad hack IMO. "Balance" should be divisible by a number. If it's not true we should address the root cause.

Reply:
Discussed with Alex, and we cannot find a better way of doing this. Casting it into Balance means our datatype remains flexible. We will keep as is